packages/block-library/src/post-featured-image/edit.js

```diff
@@ -55,14 +102,36 @@ function PostFeaturedImageDisplay( {
 					/>
 				</PanelBody>
 			</InspectorControls>
+			<BlockControls>
+				{ !! media && (
+					<ToolbarGroup>
+						<ToolbarButton
+							name="unset"
+							onClick={ () => setFeaturedImage( 0 ) }
+						>
+							{ __( 'Clear' ) }
+						</ToolbarButton>
+						<MediaReplaceFlow
```

Comment:
One thing I noticed is that `MediaReplaceFlow` uses `ToolbarGroup` internally:

https://github.com/WordPress/gutenberg/blob/e13857d4efbeef4f418b7f5c69f0ea7cf6533ae6/packages/block-editor/src/components/media-replace-flow/index.js#L120

That puts a group in another group. It something that we should look into separately as it might be too restrictive choice as @jasmussen pointed out in the linked issue.